torchao/prototype/moe_training/scaled_grouped_mm.py

```diff
@@ -195,25 +216,48 @@ def backward(ctx, grad_output: torch.Tensor):
 
         # grad_B is a special case. both operands of the grouped gemm will be 2D with offsets determing the "groups."
         # Compute scales for grad_output_t and A, which are both 2D tensors with offsets which define the "jagged" groups.
-        grad_output_t_fp8_row_major, grad_output_t_scales = (
-            triton_fp8_row_major_jagged_rowwise_scales(
-                grad_output_t_row_major,
+        if use_triton_for_per_group_scales:
+            grad_output_t_fp8_row_major, grad_output_t_scales = (
+                triton_fp8_row_major_jagged_rowwise_scales(
```

Comment:
nit yu could also do the fucntion asignment trick here to make it easier to read

Reply:
good call, done